### ISSUE TITLE
The problem with training BraTS 2023 dataset

### ISSUE THREAD

**Issue by tangqishi** [MIC-DKFZ/nnUNet#1990]
I hope to train the data for BraTS 2023. The difference between the data for 2023 and that for 2021 is that for the 2023 data, the label for the enhancing tumor is 3, whereas for the 2021 data, the label for the enhancing tumor is 4. Therefore, I made some modifications to the Dataset137_BraTS21.py file, as shown in the modifications on lines 20 and 25. After processing the data, I preprocessed it and then trained it, but during the training process, I found that the dice for ET (enhancing tumor) was always nan. I don't know why this is happening?

<img width="417" alt="input" src="https://github.com/MIC-DKFZ/nnUNet/assets/115214830/7ca50cb8-1ecb-4826-b477-105b8565a52a">
![MicrosoftTeams-image (2)](https://github.com/MIC-DKFZ/nnUNet/assets/115214830/da413486-5cef-41dd-ace3-12dee86425b2)

**Comment by tangqishi** [comment]
Also, due to my server frequently disconnecting unexpectedly, causing the need to restart model training from scratch, I need to set up checkpoints to allow the model's state to be saved at certain moments during the training process. This includes saving the model's parameters, the optimizer's state, and the learning rate scheduler's state. This way, training can resume from the last saved state instead of starting over. I have found some code that could solve this problem, but since the `nnunet` functions are quite complex, I am unsure how to modify the source code to implement this. Could you help me? 
# Optionally resume from a checkpoint
    to_restore = {"epoch": 0, "best_acc": 0.}
    utils.restart_from_checkpoint(
        os.path.join(args.output_dir, "checkpoint.pth.tar"),
        run_variables=to_restore,
        state_dict=seg_decoder,
        optimizer=optimizer,
        scheduler=scheduler,
    )
    start_epoch = to_restore["epoch"]
    best_acc = to_restore["best_acc"]  save_dict = {
                "epoch": epoch + 1,
                "state_dict": seg_decoder.state_dict(),
                "optimizer": optimizer.state_dict(),
                "scheduler": scheduler.state_dict(),
                "best_acc": best_acc,
            }
            torch.save(save_dict, os.path.join(args.output_dir, "checkpoint.pth.tar"))

**Comment by film-out** [comment]
seg_new is created with np.zeros_like so label 3 (for enhancing tumor) has to be reassigned too which is still missing in your code.
seg_new[img_npy == 3] = 3
I had the same nan dice issue for ET and this solved it.

**Comment by TSbme** [comment]
@film-out Yes, I can train it successfully. Thank you!

**Comment by liyang19971022** [comment]
@TSbme Can I ask if training BraTS2023 is fast? Why do I keep stopping at epoch 0

**Comment by TSbme** [comment]
Hi @liyang19971022,
I trained for 1000 epochs. The speed wasn't fast, but it didn't continuously stay at epoch 0. You can check the `nnUNet_results` folder for the automatically generated `training_log_2024_3_8.txt` file under that task, to see if it is indeed stuck at 0, or if it's a display issue with the server. If it really is stuck at 0, consider using multiple GPUs for task debugging.

**Comment by CodeHarcourt** [comment]
> ![Mi](https://private-user-images.githubusercontent.com/115214830/310515619-621750f9-7789-4c32-b08d-7a2597750d77.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA1NzQ4NTAsIm5iZiI6MTcxMDU3NDU1MCwicGF0aCI6Ii8xMTUyMTQ4MzAvMzEwNTE1NjE5LTYyMTc1MGY5LTc3ODktNGMzMi1iMDhkLTdhMjU5Nzc1MGQ3Ny5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxNlQwNzM1NTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wOGVjM2JjMDgzYjY1ZTQwMDVlNTJjYzcxMmEwNDcxNjRlZWVjM2YxMDhjNGQ3ZDAwYjQ2YjczZWZkNWMyNzVkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.eb0lLtUOeSBJGcf4pjdUH3gqmr0hSbWT-TNuAPC0z6M)

Hello, Are you the train_loss and the val_loss was increasing?
